code/modules/reagents/chemistry/reagents/alcohol_reagents.dm

```diff
@@ -77,6 +77,14 @@ All effects don't start immediately, but rather get worse over time; the rate is
 				// +10% success propability on each step, useful while operating in less-than-perfect conditions
 	return ..()
 
+/datum/reagent/consumable/ethanol/on_mob_add(mob/living/L,amount)
+	if(HAS_TRAIT(L, TRAIT_NO_ALCOHOL) && boozepwr > 0)
+		if(iscarbon(L))
+			var/mob/living/carbon/C = L
+			C.vomit()
+		L.reagents.remove_all_type(/datum/reagent/consumable/ethanol, amount*4, 0, 1)
+	return ..()
```

Comment:
Ever played a flyperson in the past just to annoy the heck out of everyone by vomiting all around the station? Well, this feels quite the same, except it stuns the owner.
What about dealing toxin damage on species/handle_chemicals() with an odd chance of vomiting (it clears the mob's reagents), diminished "metabolization rate" and return value TRUE?

Reply:
Vomiting makes you dry heave if you're low on nutrition, though? Flies can eat vomit so you can sorta just do that in a loop.

Also, vomiting didn't *seem* to remove reagents, I thought I checked pretty thoroughly.

Also, OOC main reason I want this is to give immediate, heavy feedback that my character doesn't want alcohol and not to give it, so making it more subtle isn't really the intent?

Reply:
Umh, a little extreme but gotcha.